llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1446,23 +1454,39 @@ bool VPlanTransforms::tryAddExplicitVectorLength(VPlan &Plan) {
 
   for (VPValue *HeaderMask : collectAllHeaderMasks(Plan)) {
     for (VPUser *U : collectUsersRecursively(HeaderMask)) {
-      auto *MemR = dyn_cast<VPWidenMemoryRecipe>(U);
-      if (!MemR)
+      VPRecipeBase *NewRecipe = nullptr;
+      auto *CurRecipe = dyn_cast<VPRecipeBase>(U);
+      if (!CurRecipe || CurRecipe->getNumDefinedValues() > 1)
```

Comment:
why is the NumDefinedValues check needed?

Reply:
Because not all recipes are single def, for example, VPInterleaveRecipe.

Reply:
Right, but why does it matter? Any recipe that isn't handled below will be skipped so there should be no need to check # of defined values unless I am missing something?

Reply:
706c6ed0d2bafba5ff54506f13cbc11c57429896
Make sense. I replaced it with assertion.